Fix recently broken RST markup

kellnerd opened the pull request:
So my observations from https://github.com/metabrainz/bookbrainz-dev-docs/pull/13#issuecomment-1460162195 were indeed not just an issue with the GitHub preview, apparently RTD does not allow you to use level 4 headings directly after a level 2 heading.
These headings are currently completely missing from https://bookbrainz-dev-docs.readthedocs.io/en/latest/docs/installation.html#manual-installation, so I've turned them into level 3 headings.

I've also fixed some broken markup which originates from #11 and dropped two references which are now unused.

Hopefully I've now fixed the most important warnings from the [latest build](https://readthedocs.org/projects/bookbrainz-dev-docs/builds/19761954/). Fingers crossed that everything works as expected this time, if not I will finally try to setup a local RTD instance.

kellnerd committed:
Correct level of headings

Previously they were skipped by RTD because they were one level too low.

kellnerd committed:
Convert broken Markdown to reStructuredText

kellnerd committed:
Use unique target names `Docker` and `Docker extension`

Duplicate target names lead to broken references.

kellnerd committed:
Drop unused references

Removed in 954b743d112a5572f173f5e850ae3593086fa010 & 5a437b58b95350f0dc09401d7430e9fb64c069b9

MonkeyDo reviewed (commented):
Thanks  for the fixes! I see some more warning in the logs when running a build locally,  some of which are related to the installation.rst file.
Logs can be found here: https://gist.github.com/MonkeyDo/22d677f72afa9325e51dedf0562f3abc

Would you mind looking at them and fixing those about installation.rst while you're fixing the file?
If you don't have time right now, we can also save it for another PR, I don't think these warnings are of much importance

kellnerd committed:
Resolve duplicate target name / label and indentation warnings

kellnerd committed:
Format literal code blocks consistently

https://www.sphinx-doc.org/en/master/usage/restructuredtext/basics.html#literal-blocks

kellnerd commented:
Sure, I think I should have silenced the remaining four warnings with 08f01bd0a24d35bcf4abbfca7a3cf5d141f58d1c.
As I was already at it, I've also formatted the remaining literal code blocks consistently in 6e5b16347c24797b2cf4a84b973a7d6d06c49b43.

MonkeyDo reviewed (approved):
Thank you!
Sorry I forgot this was in progress :)
